pkg/xds/experimental/client/client.go

```diff
@@ -0,0 +1,621 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+
+	"golang.org/x/exp/maps"
```

Comment:
we are planning to retire this x/exp/maps package (ref https://github.com/cilium/cilium/pull/34649), it would be nice to change to maps and slices package from std library.

Reply:
I couldn't use stdlib, because `go.mod` was still set to `go1.22`. I changed it now